File: rocket-bi-web/di-core/data-ingestion/domain/response/ShopifySource.ts
/*
 * @author: tvc12 - Thien Vi
 * @created: 6/1/21, 2:17 PM
 */

import { DataSourceType, DataSources } from '@core/data-ingestion';
import { DataSource } from '@core/data-ingestion/domain/response/DataSource';
import { SourceId } from '@core/common/domain';

export class ShopifySource implements DataSource {
  id: SourceId;
  orgId: string;
  databaseType: DataSourceType;
  displayName: string;
  apiUrl: string;
  apiVersion: string;
  accessToken: string;
  lastModify: number;
  creatorId: string;
  readonly className = DataSources.ShopifySource;

  constructor(
    id: SourceId,
    orgId: string,
    databaseType: DataSourceType,
    displayName: string,
    apiUrl: string,
    apiVersion: string,
    accessToken: string,
    creatorId: string,
    lastModify: number
  ) {
    this.id = id;
    this.orgId = orgId;
    this.displayName = displayName;
    this.databaseType = databaseType;
    this.apiUrl = apiUrl;
    this.apiVersion = apiVersion;
    this.accessToken = accessToken;
    this.creatorId = creatorId;
    this.lastModify = lastModify;
  }

  static fromObject(obj: any): ShopifySource {
    return new ShopifySource(obj.id, obj.orgId, obj.databaseType, obj.displayName, obj.apiUrl, obj.apiVersion, obj.accessToken, obj.creatorId, obj.lastModify);
  }
}

File: rocket-bi-server/src/main/scala/co/datainsider/jobscheduler/domain/source/ShopifySource.scala
package co.datainsider.jobscheduler.domain.source

import co.datainsider.jobscheduler.domain.Ids.SourceId
import co.datainsider.jobscheduler.domain.source.DataSourceType.DataSourceType

case class ShopifySource(
    orgId: Long = -1,
    id: SourceId,
    displayName: String,
    apiUrl: String,
    accessToken: String,
    apiVersion: String,
    creatorId: String = "",
    lastModify: Long = System.currentTimeMillis()
) extends DataSource {

  override def getId: SourceId = id

  override def getName: String = displayName

  override def getType: DataSourceType = DataSourceType.Shopify

  override def getConfig: Map[String, Any] =
    Map("api_url" -> apiUrl, "access_token" -> accessToken, "api_version" -> apiVersion)

  override def getCreatorId: String = creatorId

  override def getLastModify: Long = lastModify
}
